wsd/ClientSession.cpp

```diff
@@ -3353,30 +3471,31 @@ bool ClientSession::isTileInsideVisibleArea(const TileDesc& tile) const
 // 1. ChildSession::getClipboard() where the data for various
 //    flavours along with flavour-type and length fields are packed into the payload.
 // 2. The clipboard payload parsing code in ClipboardData::read().
-void ClientSession::preProcessSetClipboardPayload(std::string& payload)
+bool ClientSession::preProcessSetClipboardPayload(std::istream& in, std::ostream& out)
 {
-    std::size_t start = payload.find("<div id=\"meta-origin\" data-coolorigin=\"");
-    if (start != std::string::npos)
+    if (!Util::copyToMatch(in, out, "<div id=\"meta-origin\" data-coolorigin=\""))
+        return false;
+
+    // discard this tag
+    if (!Util::seekToMatch(in, "\">\n"))
     {
-        std::size_t end = payload.find("\">\n", start);
-        if (end == std::string::npos)
-        {
-            LOG_DBG("Found unbalanced starting meta <div> tag in setclipboard payload.");
-            return;
-        }
+        LOG_DBG("Found unbalanced starting meta <div> tag in setclipboard payload.");
+        return false;
+    }
+    in.seekg(3, std::ios_base::cur);
```

Comment:
Nit: I tend to go with `strlen("\">\n")` for these cases, it's optimized to 3 by gcc and it's more readable.